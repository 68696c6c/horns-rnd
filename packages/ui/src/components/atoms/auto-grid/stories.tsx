/* eslint-disable react/jsx-one-expression-per-line */
import React from 'react'
import { Story, Meta } from '@storybook/react/types-6-0'

import { Breakpoint } from '../../../config'
import { AutoGrid, AutoGridProps } from '.'

export default {
  title: 'Atoms/AutoGrid',
  component: AutoGrid,
} as Meta

const Template: Story<AutoGridProps> = ({
  breakpoint,
  children,
  ...others
}: AutoGridProps) => (
  <>
    <p>
      Above the {breakpoint} breakpoint, these items will rendered in an
      auto-sized, wrapping grid.
    </p>
    <AutoGrid breakpoint={breakpoint} {...others}>
      {children}
    </AutoGrid>
  </>
)

export const Default = Template.bind({})
Default.args = {
  breakpoint: Breakpoint.Mobile,
  children: (
    <>
      <div>one</div>
      <div>two</div>
      <div>three</div>
      <div>four</div>
      <div>five</div>
      <div>six</div>
    </>
  ),
}

export const Gapped = Template.bind({})
Gapped.args = {
  breakpoint: Breakpoint.Mobile,
  gapped: true,
  children: (
    <>
      <div>one</div>
      <div>two</div>
      <div>three</div>
      <div>four</div>
      <div>five</div>
      <div>six</div>
    </>
  ),
}

export const SingleItem = Template.bind({})
SingleItem.args = {
  breakpoint: Breakpoint.Mobile,
  children: <div>one</div>,
}
