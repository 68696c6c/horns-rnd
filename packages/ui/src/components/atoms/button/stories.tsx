import React from 'react'
import { Meta, Story } from '@storybook/react/types-6-0'

import { clickableButtonDemo } from '../../../_story'
import { Colorway, Font, Size } from '../../../config'
import { ButtonProps } from '../../quarks'

import { Button } from '.'

export default {
  title: 'Atoms/Button',
  component: Button,
} as Meta

const Template: Story<ButtonProps> = ({ children, ...others }: ButtonProps) => (
  <Button {...others}>{children}</Button>
)

export const Default = Template.bind({})
Default.args = {
  children: 'Button',
}

export const Props = Template.bind({})
Props.args = {
  ...Default.args,
  border: { all: { width: Size.Small } },
  color: Colorway.Primary,
  padding: { all: Size.Small },
  radius: { all: Size.Large },
  font: Font.Emphasized,
}

export const Traits = () => clickableButtonDemo(Default, { ...Default.args })
