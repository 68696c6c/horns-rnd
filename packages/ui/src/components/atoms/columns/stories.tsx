/* eslint-disable react/jsx-one-expression-per-line */
import React from 'react'
import { Story, Meta } from '@storybook/react/types-6-0'

import { Breakpoint } from '@horns/theme'

import { StyledCode } from '../../../_story'

import { Columns, ColumnsProps } from '.'

export default {
  title: 'Atoms/Columns',
  component: Columns,
} as Meta

const Template: Story<ColumnsProps> = ({
  breakpoint,
  columns,
  children,
  ...others
}: ColumnsProps) => (
  <>
    <p>
      Above the {breakpoint} breakpoint, these items will be rendered as
      columns. By default, a single row of columns will be rendered with each
      item in its own column. The <StyledCode>columns</StyledCode> prop will
      limit the number of columns per row and create additional rows to fit the
      remaining items.
    </p>
    <Columns breakpoint={breakpoint} columns={columns} {...others}>
      {children}
    </Columns>
  </>
)

export const Default = Template.bind({})
Default.args = {
  breakpoint: Breakpoint.Mobile,
  children: (
    <>
      <div>one</div>
      <div>two</div>
      <div>three</div>
      <div>four</div>
      <div>five</div>
      <div>six</div>
    </>
  ),
}

export const Gapped = Template.bind({})
Gapped.args = {
  breakpoint: Breakpoint.Mobile,
  gapped: true,
  children: (
    <>
      <div style={{ background: 'tan' }}>one</div>
      <div style={{ background: 'tan' }}>two</div>
      <div style={{ background: 'tan' }}>three</div>
      <div style={{ background: 'tan' }}>four</div>
      <div style={{ background: 'tan' }}>five</div>
      <div style={{ background: 'tan' }}>six</div>
    </>
  ),
}

export const SingleItem = Template.bind({})
SingleItem.args = {
  breakpoint: Breakpoint.Mobile,
  children: <div>one</div>,
}

export const FixedColumns = Template.bind({})
FixedColumns.args = {
  breakpoint: Breakpoint.Mobile,
  columns: 4,
  children: (
    <>
      <div>one</div>
      <div>two</div>
      <div>three</div>
      <div>four</div>
      <div>five</div>
      <div>six</div>
    </>
  ),
}
