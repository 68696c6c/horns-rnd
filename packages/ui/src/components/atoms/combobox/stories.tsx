/* eslint-disable react/jsx-one-expression-per-line,jsx-a11y/label-has-associated-control */
import React from 'react'
import { Story, Meta } from '@storybook/react/types-6-0'

import { Combobox, ComboboxProps } from '.'

export default {
  title: 'Atoms/Combobox',
  component: Combobox,
} as Meta

const options = [
  { label: 'One', value: '1' },
  { label: 'Two', value: '2' },
  { label: 'Three', value: '3' },
  { label: 'Four', value: '4' },
  { label: 'Five', value: '5' },
]

const Template: Story<ComboboxProps> = (props: ComboboxProps) => (
  <Combobox {...props} />
)

export const Default = Template.bind({})
Default.args = {
  id: 'example-select',
  name: 'example_select',
  placeholder: 'Select one...',
  options,
}

export const Multiselect = Template.bind({})
Multiselect.args = {
  ...Default.args,
  id: 'example-multiselect',
  name: 'example_multiselect',
  placeholder: 'Select one or more...',
  multiple: true,
}
