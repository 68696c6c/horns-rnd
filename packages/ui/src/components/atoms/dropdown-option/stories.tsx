import React from 'react'
import { Story, Meta } from '@storybook/react/types-6-0'

import { DropdownOption } from '.'

export default {
  title: 'Atoms/DropdownOption',
  component: DropdownOption,
} as Meta

export const Default: Story = () => <DropdownOption>hello world</DropdownOption>
