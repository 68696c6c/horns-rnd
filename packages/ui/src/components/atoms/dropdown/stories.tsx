import React from 'react'
import { Story, Meta } from '@storybook/react/types-6-0'

import { Dropdown } from '.'

export default {
  title: 'Atoms/Dropdown',
  component: Dropdown,
} as Meta

export const Default: Story = () => (
  <Dropdown open>
    <li>item one</li>
    <li>item two</li>
    <li>item three</li>
  </Dropdown>
)
