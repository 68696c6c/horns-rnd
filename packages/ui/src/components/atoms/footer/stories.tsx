import React from 'react'
import { Story, Meta } from '@storybook/react/types-6-0'

import { Colorway, Size } from '@horns/theme'

import { blockDemo } from '../../../_story'
import { BlockProps } from '../../quarks'

import { Footer } from '.'

export default {
  title: 'Atoms/Footer',
  component: Footer,
} as Meta

const Template: Story<BlockProps> = ({ children, ...others }: BlockProps) => (
  <Footer {...others}>{children}</Footer>
)

export const Default = Template.bind({})
Default.args = {
  children: 'Footer',
}

export const Props = Template.bind({})
Props.args = {
  ...Default.args,
  color: Colorway.Primary,
  padding: { all: Size.Small },
  fluid: false,
}

export const Traits = () => blockDemo(Footer, {})
