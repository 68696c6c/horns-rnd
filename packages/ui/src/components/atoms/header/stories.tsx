import React from 'react'
import { Story, Meta } from '@storybook/react/types-6-0'

import { blockDemo } from '../../../_story'
import { Colorway, Size } from '../../../config'
import { BlockProps } from '../../quarks'

import { Header } from '.'

export default {
  title: 'Atoms/Header',
  component: Header,
} as Meta

const Template: Story<BlockProps> = ({ children, ...others }: BlockProps) => (
  <Header {...others}>{children}</Header>
)

export const Default = Template.bind({})
Default.args = {
  children: 'Header',
}

export const Props = Template.bind({})
Props.args = {
  ...Default.args,
  color: Colorway.Primary,
  padding: { all: Size.Small },
  fluid: false,
}

export const ResponsiveMobile = Template.bind({})
ResponsiveMobile.args = {
  ...Default.args,
  fluid: false,
}
ResponsiveMobile.parameters = {
  viewport: {
    defaultViewport: 'mobile1',
  },
}

export const ResponsiveMax = Template.bind({})
ResponsiveMax.args = {
  ...Default.args,
  fluid: false,
}
ResponsiveMax.parameters = {
  viewport: {
    defaultViewport: 'max',
  },
}

export const ResponsiveFluid = Template.bind({})
ResponsiveFluid.args = {
  ...Default.args,
  fluid: true,
}
ResponsiveFluid.parameters = {
  viewport: {
    defaultViewport: 'max',
  },
}

export const Traits = () => blockDemo(Header, {})
