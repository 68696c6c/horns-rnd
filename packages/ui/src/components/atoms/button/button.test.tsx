import React from 'react'

import { assertStateStyles, render, screen } from '../../test'

import { Default, Props } from './stories'

describe('Button', () => {
  it('should render as default', () => {
    const { container } = render(<Default {...Default.args}>example</Default>)
    expect(container).toMatchSnapshot()
  })
  it('should support trait props', () => {
    const { container } = render(<Props {...Props.args}>example</Props>)
    expect(container).toMatchSnapshot()
  })
  describe('ui states', () => {
    it('should render default state', () => {
      render(<Default>default</Default>)
      assertStateStyles(screen.getByRole('button'), '', {
        'background-color': 'rgb(255, 255, 255)',
      })
    })
    it('should render hover state', () => {
      render(<Default>hover</Default>)
      assertStateStyles(screen.getByRole('button'), ':hover', {
        'background-color': 'rgb(247, 247, 247)',
      })
    })
    it('should render active state', () => {
      render(<Default>active</Default>)
      assertStateStyles(screen.getByRole('button'), ':active', {
        'background-color': 'rgb(240, 240, 240)',
      })
    })
  })
})
