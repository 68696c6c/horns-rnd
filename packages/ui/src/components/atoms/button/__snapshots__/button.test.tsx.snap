// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render as default 1`] = `
.emotion-0 {
  display: inline-block;
  border-color: rgb(224, 224, 224);
  background-color: rgb(255, 255, 255);
  color: rgb(1, 1, 1);
  border-top-width: 2px;
  border-top-style: solid;
  border-right-width: 2px;
  border-right-style: solid;
  border-bottom-width: 2px;
  border-bottom-style: solid;
  border-left-width: 2px;
  border-left-style: solid;
  cursor: pointer;
  padding-top: 12px;
  padding-right: 16px;
  padding-bottom: 12px;
  padding-left: 16px;
  border-top-left-radius: 2px;
  border-top-right-radius: 2px;
  border-bottom-left-radius: 2px;
  border-bottom-right-radius: 2px;
  font-family: Helvetica,sans-serif;
  font-kerning: normal;
  font-size: medium;
  font-style: normal;
  font-weight: 700;
  line-height: 1.6em;
  text-align: left;
  text-decoration-line: none;
  text-transform: none;
}

.emotion-0:disabled,
.emotion-0.inactive {
  border-color: rgb(165, 165, 165);
  background-color: rgb(191, 191, 191);
  color: rgb(1, 1, 1);
}

.emotion-0:not(:disabled):not(.inactive):hover,
.emotion-0:not(:disabled):not(.inactive).hover {
  border-color: rgb(224, 224, 224);
  background-color: rgb(247, 247, 247);
  color: rgb(1, 1, 1);
}

.emotion-0:not(:disabled):not(.inactive):active,
.emotion-0:not(:disabled):not(.inactive).active {
  border-color: rgb(224, 224, 224);
  background-color: rgb(240, 240, 240);
  color: rgb(1, 1, 1);
}

.emotion-0:not(:disabled):not(.inactive):focus,
.emotion-0:not(:disabled):not(.inactive).focus {
  outline-style: dotted;
  outline-color: inherit;
  outline-width: 1px;
  outline-offset: 1px;
}

<div>
  <button
    class="emotion-0"
  >
    example
  </button>
</div>
`;

exports[`Button should support trait props 1`] = `
.emotion-0 {
  display: inline-block;
  border-color: rgb(204, 136, 0);
  background-color: rgb(255, 170, 0);
  color: rgb(1, 1, 1);
  border-top-width: 12px;
  border-top-style: solid;
  border-right-width: 12px;
  border-right-style: solid;
  border-bottom-width: 12px;
  border-bottom-style: solid;
  border-left-width: 12px;
  border-left-style: solid;
  cursor: pointer;
  padding-top: 12px;
  padding-right: 16px;
  padding-bottom: 12px;
  padding-left: 16px;
  border-top-left-radius: 24px;
  border-top-right-radius: 24px;
  border-bottom-left-radius: 24px;
  border-bottom-right-radius: 24px;
  font-family: Helvetica,sans-serif;
  font-kerning: normal;
  font-size: medium;
  font-style: italic;
  font-weight: 400;
  line-height: 1.6em;
  text-align: left;
  text-decoration-line: none;
  text-transform: none;
}

.emotion-0:disabled,
.emotion-0.inactive {
  border-color: rgb(165, 140, 89);
  background-color: rgb(191, 149, 64);
  color: rgb(1, 1, 1);
}

.emotion-0:not(:disabled):not(.inactive):hover,
.emotion-0:not(:disabled):not(.inactive).hover {
  border-color: rgb(255, 170, 0);
  background-color: rgb(204, 136, 0);
  color: rgb(1, 1, 1);
}

.emotion-0:not(:disabled):not(.inactive):active,
.emotion-0:not(:disabled):not(.inactive).active {
  border-color: rgb(204, 136, 0);
  background-color: rgb(135, 90, 0);
  color: rgb(255, 255, 255);
}

.emotion-0:not(:disabled):not(.inactive):focus,
.emotion-0:not(:disabled):not(.inactive).focus {
  outline-style: dotted;
  outline-color: inherit;
  outline-width: 1px;
  outline-offset: 1px;
}

<div>
  <button
    class="emotion-0"
    color="primary"
    radius="[object Object]"
  >
    example
  </button>
</div>
`;
